### PR TITLE
Expect `material3.ModalBottomSheet` in common

### PR DESCRIPTION
**Pull request by MatkovIvan**
## Proposed Changes
- Expect `ModalBottomSheet` in common and implement for skiko
- Add test usage in mpp demo app

Screenshot 2023-09-06 at 18 15 15

# API Changes

```diff
+@Composable
+@ExperimentalMaterial3Api
+expect fun ModalBottomSheet(
+    onDismissRequest: () -> Unit,
+    modifier: Modifier = Modifier,
+    sheetState: SheetState = rememberModalBottomSheetState(),
+    shape: Shape = BottomSheetDefaults.ExpandedShape,
+    containerColor: Color = BottomSheetDefaults.ContainerColor,
+    contentColor: Color = contentColorFor(containerColor),
+    tonalElevation: Dp = BottomSheetDefaults.Elevation,
+    scrimColor: Color = BottomSheetDefaults.ScrimColor,
+    dragHandle: @Composable (() -> Unit)? = { BottomSheetDefaults.DragHandle() },
+    windowInsets: WindowInsets = BottomSheetDefaults.windowInsets,
+    content: @Composable ColumnScope.() -> Unit,
+)
```

## Testing

Test: Try to use `ModalBottomSheet` from common or look at mpp demo

## Issues Fixed

https://github.com/JetBrains/compose-multiplatform/issues/3528

### CODE REVIEW

File: compose/material3/material3/src/skikoMain/kotlin/androidx/compose/material3/ModalBottomSheet.skiko.kt [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+ * @param tonalElevation The tonal elevation of this bottom sheet.
+ * @param scrimColor Color of the scrim that obscures content when the bottom sheet is open.
+ * @param dragHandle Optional visual marker to swipe the bottom sheet.
+ * @param windowInsets window insets to be passed to the bottom sheet window via [PaddingValues]
+ * params.
+ * @param content The content to be displayed inside the bottom sheet.
+ */
+@ExperimentalMaterial3Api
+@Composable
+actual fun ModalBottomSheet(
```
Comment (igordmn) [inline compose/material3/material3/src/skikoMain/kotlin/androidx/compose/material3/ModalBottomSheet.skiko.kt]: It seems it is a copy of Android implementation except the Popup part.

As an option, we can move it to `commonMain`, and make only the Popup part actual. It should look like that if we would commonize it in AOSP. Not sure what is better for the fork:
- copy Android -> no merge conflict, but no knowledge of the new features
- move Android to common -> merge conflict (or not, if git will treat this as a rename), knowledge of the new features

**Review by igordmn — APPROVED** [review]

File: compose/material3/material3/src/skikoMain/kotlin/androidx/compose/material3/ModalBottomSheet.skiko.kt (hunk 1 above)
Comment (MatkovIvan) [reply compose/material3/material3/src/skikoMain/kotlin/androidx/compose/material3/ModalBottomSheet.skiko.kt]: > It seems it is a copy of Android implementation except the Popup part.

Yes, but not exactly. Android custom Popup adds inset as parameter, so I had to apply it explicitly.
So, yes, more code can be moved to common if we want (requires introducing more changes in android part and some internal expect/actual). Not a big deal for me.

@m-sasha thoughts?

Comment (m-sasha) [reply compose/material3/material3/src/skikoMain/kotlin/androidx/compose/material3/ModalBottomSheet.skiko.kt]: If it's not _significantly_ more work, I think it's better to commonize more.

**Review by m-sasha — APPROVED** [review]

File: compose/material3/material3/src/skikoMain/kotlin/androidx/compose/material3/ModalBottomSheet.skiko.kt (hunk 1 above)
Comment (MatkovIvan) [reply compose/material3/material3/src/skikoMain/kotlin/androidx/compose/material3/ModalBottomSheet.skiko.kt]: Done

**Review by igordmn — APPROVED** [review]

**Pull request merged**